### PR TITLE
feat(scss) use grey palette for disabled state on framed form components

### PR DESCRIPTION
**Pull request by MarcFairbrother**
## Description

- uses grey palette for disabled state on framed form components
- adds story for framed forms

-----

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-1490

File: packages/scss/src/mixins/_forms.scss [hunk 1]
```diff
@@ -1,33 +1,35 @@
 @import "../mixins/utilities";
 
 @mixin fieldState($fieldname, $state) {
-	background-color: _color($state, "50");
+	&:not(:disabled) {
```
Comment (AlexiaAlarza) [inline packages/scss/src/mixins/_forms.scss:4]: &:not(:disabled):not(.is-disabled) à rajouter ?

File: packages/scss/src/components/inputs/textfield/_textfield.framed.scss [hunk 2]
```diff
@@ -60,74 +60,23 @@
...
-							}
-						}
-					}
-				}
-			}
-
-			&.is-error {
-				box-shadow: fakeborderoverlay(_color("error"));
-				z-index: 3;
+		&:not(:disabled, .is-error, .is-success, .is-valid, .is-warning, .is-invalid, .ng-invalid) {
```
Comment (AlexiaAlarza) [inline packages/scss/src/components/inputs/textfield/_textfield.framed.scss:67]: &:not(:disabled):not(.is-disabled) à rajouter ?

File: packages/scss/src/components/_form.scss [hunk 3]
```diff
@@ -139,7 +140,7 @@
 		clear: both;
 	}
 
-	&:hover {
+	&:not(.is-disabled):hover {
```
Comment (AlexiaAlarza) [inline packages/scss/src/components/_form.scss:143]: &:not(:disabled):not(.is-disabled) à rajouter ?

Comment (MarcFairbrother) [reply packages/scss/src/components/_form.scss:143]: This `.is-disabled` class is used on a parent of the `<input>`, not on the input itself so we cannot target the `disabled` attribute.

File: packages/scss/src/components/inputs/textfield/_textfield.framed.scss (hunk 2 above)
Comment (MarcFairbrother) [reply packages/scss/src/components/inputs/textfield/_textfield.framed.scss:67]: The `.is-disabled` class doesn't work on the framed form elements (styles are overridden by the more specific framed style declarations). I feel like this is a bad practice in any case and the `.is-disabled` class should be deprecated in favour of the `disabled` attribute.

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-1490

**Review by AlexiaAlarza — APPROVED** [review]

**Pull request merged**